### ISSUE TITLE
Notification bar doesn't update

### ISSUE THREAD

**Issue by InstallGentoo11** [vanilla-music/vanilla#1151]
- Vanilla Music Version (Settings -> About): 1.2.2
- Android Version: 12 
- Phone vendor / model: Pixel 4a 5G
- Application was downloaded from (use `[x]`):
  - [ ] Play Store
  - [x] F-Droid
  - [ ] Other (Nightly Build)
- Did this work in a previous version?
  - [ ] Yes
  - [ ] No
  - [x] NA / Don't know.

## Description

In the notification bar or the lock screen the popup thingy doesn't display the correct song information (name, art, ect), it is one song behind, except the first song, It has been doing this for a while now. I tried even nuking all the data related to the app. If I pause the song it seems to update the information. Pretty sure it didn't use to do this before the nuking the app.

**Comment by adrian-bl** [comment]
I played with this and this looks very much like a platform bug / race condition.

Vanilla correctly sets the metadata, but Android sometimes decides to show old data. The issue goes almost away if i disable cover support - but is still happening *sometimes*.

As mentioned, this only happens for me in an emulator and not on real hardware - and i'm not sure what we can do to work around this.

**Comment by eylenburg** [comment]
I can confirm this bug. Vanilla 1.3.0 on a Samsung S20 with Android 13. Doesn't happen on other music apps

**Comment by reactorcoremeltdown** [comment]
Can also confirm, running on Motorola G52.

BTW that only started to happen recently. Might try to drop app cache first; if that does not help, I'll take a look on how it is implemented in other apps.

**Comment by reactorcoremeltdown** [comment]
Nope, wiping the cache didn't help at slightest. Indeed could be a platform race condition, or something. Will check how other players update notification area.

**Comment by adrian-bl** [comment]
Wiping the cache is very unlikely to help with most issues.

This is defenitively a platform bug: tidal is hitting this quite often on my phone running los20

**Comment by eylenburg** [comment]
Still have this problem on Vanilla 1.3.1 / Android 14 GrapheneOS / Pixel 8

**Comment by Ahapulin** [comment]
> - Vanilla Music Version (Settings -> About): 1.2.2
> - Android Version: 12 
> - Phone vendor / model: Pixel 4a 5G
> - Application was downloaded from (use `[x]`):
>   - [ ] Play Store
>   - [x] F-Droid
>   - [ ] Other (Nightly Build)
> - Did this work in a previous version?
>   - [ ] Yes
>   - [ ] No
>   - [x] NA / Don't know.
> 
> ## Description
> 
> In the notification bar or the lock screen the popup thingy doesn't display the correct song information (name, art, ect), it is one song behind, except the first song, It has been doing this for a while now. I tried even nuking all the data related to the app. If I pause the song it seems to update the information. Pretty sure it didn't use to do this before the nuking the app.

**Comment by reactorcoremeltdown** [comment]
@adrian-bl apologies for a tag, but I think it would ease the pain of Vanilla users.

A month into testing of the new PR yields a good result, I can see that the notification updates promptly. Would you be able to provide a review, please?

Thanks in advance!

**Comment by JetpackJackson** [comment]
I'm having this issue on Android 14, Samsung S21 5G, stock rom. Is there a beta build/patch available to test, out of curiosity? Thanks in advance, and thanks for the app! I'm liking it way more than VLC (even with this minor glitch lol!)

**Comment by reactorcoremeltdown** [comment]
> I'm having this issue on Android 14, Samsung S21 5G, stock rom. Is there a beta build/patch available to test, out of curiosity? Thanks in advance, and thanks for the app! I'm liking it way more than VLC (even with this minor glitch lol!)

You might want to try a recent build of my fork where I test my experimental features.

https://fdroid.tiredsysadmin.cc/repo/space.rcmd.android.vanillaplus_13121.apk

Proceed at your own risk! This is a different application with a different icon and id, so you'll need to scan your media folders and adjust your settings from scratch!

It is a build where I am testing offline Rockbox-style scrobbling, and there's currently no flag or setting to disable it (will work on that & provide upstream port whenever I have time).

**Comment by JetpackJackson** [comment]
Alright! I tested it out and I encountered the issue again but it only happened once vs it happening for each song.

**Comment by adrian-bl** [comment]
I wonder if it would help if we send the notification in a background thread (handler).

Or another hack: send it without cover art first and then with

**Comment by JetpackJackson** [comment]
Possibly, those might work!
